src/node.cc

```diff
@@ -4271,6 +4264,17 @@ static void StartNodeInstance(void* arg) {
     Local<Context> context = Context::New(isolate);
     Context::Scope context_scope(context);
     Environment env(&isolate_data, context);
+
+    uv_async_init(env.event_loop(),
+                  env.dispatch_debug_messages_async(),
+                  DispatchDebugMessagesAsyncCallback);
+    uv_unref(reinterpret_cast<uv_handle_t*>(
+             env.dispatch_debug_messages_async()));
+
+    if (!instance_data->use_debug_agent()) {
+      RegisterDebugSignalHandler(env.dispatch_debug_messages_async());
```

Comment:
> Sorry, I just realized this is probably not going to work. It installs the signal handler (and starts the watchdog thread, initializes the debug semaphore, etc.) for every context. It's not academic: Electron creates multiple contexts.

No worries, I appreciate your help and learned for doing it. Just to clarify, is that [Electron](http://electron.atom.io/) that you are referring to?